store/store.go

```diff
@@ -1255,9 +1270,50 @@ func (s *Store) downloadDelta(name string, downloadDir string, downloadInfo *sna
 		return "", err
 	}
 
+	// TODO: Check sha3_384
 	return deltaPath, nil
 }
 
+// applyDelta generates a target snap from the current snap and a downloaded delta.
+var applyDelta = func(name string, deltaPath string, deltaInfo *snap.DeltaInfo) (string, error) {
```

Comment:
I guess you do this for mocking? You may consider using "testutils.MockCommand("xdelta")" and leave this as a real function. Then you can check the "xdelta" command mock arguments and also simulate failure via the MockCommand